viz/pkg/healthcheck/healthcheck.go

```diff
@@ -54,147 +73,242 @@ func (hc *HealthChecker) RunChecks(observer healthcheck.CheckObserver) bool {
-				pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, hc.vizNamespace)
-				if err != nil {
-					return err
-				}
+					identityName := fmt.Sprintf("linkerd-tap.%s.svc", hc.vizNamespace)
+					return hc.CheckCertAndAnchors(cert, anchors, identityName)
+				}),
+			*healthcheck.NewChecker("tap API server cert is valid for at least 60 days").
+				WithHintAnchor("l5d-webhook-cert-not-expiring-soon").
+				Warning().
+				WithCheck(func(ctx context.Context) error {
+					cert, err := hc.FetchCredsFromSecret(ctx, hc.vizNamespace, tapTLSSecretName)
+					if kerrors.IsNotFound(err) {
+						cert, err = hc.FetchCredsFromOldSecret(ctx, hc.vizNamespace, tapOldTLSSecretName)
+					}
+					if err != nil {
+						return err
+					}
+					return hc.CheckCertAndAnchorsExpiringSoon(cert)
+				}),
+			*healthcheck.NewChecker("tap API service is running").
+				WithHintAnchor("l5d-tap-api").
+				Warning().
+				WithRetryDeadline(hc.RetryDeadline).
+				WithCheck(func(ctx context.Context) error {
+					return hc.CheckAPIService(ctx, linkerdTapAPIServiceName)
+				}),
+			*healthcheck.NewChecker("linkerd-viz pods are injected").
+				WithHintAnchor("l5d-viz-pods-injection").
+				Warning().
+				WithCheck(func(ctx context.Context) error {
+					pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, hc.vizNamespace)
+					if err != nil {
+						return err
+					}
+					return healthcheck.CheckIfDataPlanePodsExist(pods)
+				}),
+			*healthcheck.NewChecker("viz extension pods are running").
+				WithHintAnchor("l5d-viz-pods-running").
+				Warning().
+				WithRetryDeadline(hc.RetryDeadline).
+				SurfaceErrorOnRetry().
+				WithCheck(func(ctx context.Context) error {
+					pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, hc.vizNamespace)
+					if err != nil {
+						return err
+					}
 
-				// Check for relevant pods to be present
-				err = healthcheck.CheckForPods(pods, []string{"linkerd-grafana", "linkerd-prometheus", "linkerd-web", "linkerd-tap"})
-				if err != nil {
-					return err
+					// Check for relevant pods to be present
+					err = healthcheck.CheckForPods(pods, []string{"linkerd-grafana", "linkerd-prometheus", "linkerd-web", "linkerd-tap"})
+					if err != nil {
+						return err
+					}
+
+					return healthcheck.CheckPodsRunning(pods, "")
+				}),
+			*healthcheck.NewChecker("can initialize the client").
+				WithHintAnchor("l5d-viz-existence-client").
+				Fatal().
+				WithCheck(func(ctx context.Context) (err error) {
+					hc.vizAPIClient, err = client.NewExternalClient(ctx, hc.vizNamespace, hc.KubeAPIClient())
+					return
+				}),
+			*healthcheck.NewChecker("data plane proxy metrics are present in Prometheus").
+				WithHintAnchor("l5d-data-plane-prom").
+				Warning().
+				WithRetryDeadline(hc.RetryDeadline).
+				WithCheck(func(ctx context.Context) (err error) {
+					pods, err := hc.getDataPlanePodsFromVizAPI(ctx)
+					if err != nil {
+						return err
+					}
+
+					// TODO: Check if prometheus is present
```

Comment:
Is this actually important? My thinking is that, As this query goes through viz metrics-api, This also supports external prometheus use-case.

It feels okay to have this check fail, when they have viz extension installed with no default/external prometheus as most functionality is lost in that case🤔